### PR TITLE
Connect a Slack team to a Kifi org or create one automatically

### PR DESCRIPTION
**Pull request by leogrim**
@ryanpbrewster @andrewconner @eishay

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/json/package.scala [hunk 1]
```diff
@@ -201,6 +201,17 @@ package object json {
       }
     }
   }
+
+  val formatNone: Format[None.type] = {
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/json/package.scala:205]: Hah, nice.

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes [hunk 2]
```diff
@@ -547,6 +548,9 @@ GET           /site/organizations/:id/referralCode              @com.keepit.cont
 POST          /site/organizations/:id/redeemCode                @com.keepit.controllers.website.PaymentsController.redeemCreditCode(id: PublicId[Organization])
 GET           /site/organizations/:id/rewards                   @com.keepit.controllers.website.PaymentsController.getRewards(id: PublicId[Organization])
 
+POST          /site/organizations/create/slack                  @com.keepit.controllers.website.SlackController.createOrganizationForSlackTeam(slackTeamId: String)
+POST          /site/organizations/:id/slack/connect             @com.keepit.controllers.website.SlackController.connectSlackTeamToOrganization(id: PublicId[Organization], slackTeamId: String)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:552]: Will this logic automatically happen if someone creates an integration from a org lib to slack channel? Or it's a special prompt, like "Do you want to connect _x_ with _y_?".

**Comment by andrewconner** [comment]
Looks good :+1:

File: kifi-backend/modules/shoebox/conf/shoeboxService.routes (hunk 2 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/conf/shoeboxService.routes:552]: This is totally separate. The only thing that will happen when someone creates a library integration is auto-joining the org his / her Slack team may be connected to.

File: kifi-backend/modules/shoebox/app/com/keepit/slack/SlackCommander.scala [hunk 3]
```diff
@@ -117,6 +126,11 @@ class SlackCommanderImpl @Inject() (
           lastPostedAt = None
         ))
       }
+      slackTeamRepo.getBySlackTeamId(auth.teamId).foreach { team =>
+        team.organizationId.foreach { orgId =>
+          orgMembershipCommander.unsafeAddMembership(OrganizationMembershipAddRequest(orgId, userId, userId))
+        }
+      }
     }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackCommander.scala:134]: @andrewconner here, this is it, that happens every time we persist your Slack credentials

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/DeepLinkRouter.scala [hunk 4]
```diff
@@ -115,6 +116,7 @@ object DeepLinkType {
   val InvitedLibraries = "il"
   val FriendRequest = "fr"
   val OrganizationInvite = "oi"
+  val OrganizationView = "ov"
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/DeepLinkRouter.scala:119]: We should try and remember to tell the mobile team about this at some point.

File: kifi-backend/modules/shoebox/app/com/keepit/slack/SlackCommander.scala [hunk 5]
```diff
@@ -318,6 +332,69 @@ class SlackCommanderImpl @Inject() (
...
+        slackClient.getTeamInfo(token).flatMap { teamInfo =>
+          val orgInitialValues = OrganizationInitialValues(name = teamInfo.name.value, description = None, site = teamInfo.emailDomain.map(_.value))
+          orgCommander.createOrganization(OrganizationCreateRequest(userId, orgInitialValues)) match {
+            case Right(createdOrg) =>
+              val orgId = createdOrg.newOrg.id.get
+              teamInfo.emailDomain.foreach { domain => orgDomainCommander.addDomainOwnership(OrganizationDomainAddRequest(userId, orgId, domain.value)) }
+              teamInfo.icon.maxByOpt(_._1).foreach {
+                case (size, imageUrl) =>
+                // todo(Léo): upload as org avatar
+              }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackCommander.scala:364]: @andrewconner

**Pull request merged**